### DISCUSSION TITLE
Changing color based on sensor's value

### DISCUSSION THREAD

**Discussion by jorgemarmor** [custom-cards/button-card#870] [Q&A]
Hi, 

I'm trying to change the color of the button based on a sensors' value. I've been reading around, but I couldn't make it to work. Right now I'm using the following code:

```
type: custom:button-card
coloype: custom:button-card
color_type: icon
entity: input_button.reiniciar_estigia
show_name: false
tap_action:
  action: toggle
color: |
  [[[ 
     if (sensor.ordenador_potencia > 1) return rgb(135,179,246);
     else return rgb(169,177,188); 
   ]]]
```

Any idea on what can be wrong?

Thanks in advance

**Comment by jorgemarmor** [comment]
Self response, after some more days of research, digging and trying options, so this can be useful for somebody else:

Apparently the problem with my code was a combination of:

- Color stand alone works well with rgb (r,g,b) but not with formula -> I had to call to styles: icon: color: instead.
- I struggled to get the value of the sensor. I managed to work it out with "states['name'].state
- rgb(r,g,b) wasn't working within the formula, I had to use hex code instead.

So, in summary, the following code worked:

```
type: custom:button-card
color_type: icon
entity: input_button.reiniciar_estigia
show_name: false
tap_action:
  action: toggle
styles:
  icon:
    - color: |
        [[[
          if (states['sensor.ordenador_potencia'].state > 1)  return '#87b3f6' 
          else return '#a9b1bc'
        ]]]
```
